src/guc.c

```diff
@@ -133,13 +133,14 @@ _guc_init(void)
 							"Maximum open chunks per insert",
 							"Maximum number of open chunk tables per insert",
 							&ts_guc_max_open_chunks_per_insert,
-							work_mem * 1024L / 25000L, /* Measurements via
-														* `MemoryContextStats(TopMemoryContext)`
-														* show chunk insert
-														* state memory context
-														* takes up ~25K bytes
-														* (work_mem is in
-														* kbytes) */
+							Min(work_mem * 1024L / 25000L,
```

Comment:
```suggestion
							Min(work_mem * INT64CONST(1024) / INT64CONST(25000L),
```
I think otherwise it can still overflow the multiplication